lib/core/src/Cardano/Wallet.hs

```diff
@@ -507,9 +519,48 @@ newWalletLayer tracer bp db nw tl = do
         fmap (\(PrimaryKey wid) -> wid) <$> DB.listWallets db
 
     _listTransactions
-        :: WalletId
-        -> ExceptT ErrNoSuchWallet IO [(Tx t, TxMeta)]
-    _listTransactions wid = liftIO $ map snd <$> DB.readTxHistory db (PrimaryKey wid)
+        :: (DefineTx t)
+        => WalletId
+        -> IO [TransactionInfo]
+    _listTransactions wid = assemble <$> DB.readTxHistory db (PrimaryKey wid)
+      where
+        -- This relies on DB.readTxHistory returning all necessary transactions
+        -- to assemble coin selection information for outgoing payments.
+        -- To reliably provide this information, it should be looked up when
+        -- applying blocks, but that is future work (issue #573).
+        assemble :: [(Hash "Tx", (Tx t, TxMeta))] -> [TransactionInfo]
+        assemble txs = map mkTxInfo txs
+          where
+            mkTxInfo (txid, (tx, meta)) = TransactionInfo
+                { txInfoId = txid
+                , txInfoInputs = [(txIn, lookupOutput txIn) | txIn <- W.inputs @t tx]
+                , txInfoOutputs = W.outputs @t tx
+                , txInfoMeta = meta
+                , txInfoDepth = Quantity 0
+                -- fixme: this depth is calculated in slots, but a block depth is required.
+                -- That will require tracking the block height and recording it in TxMeta.
```

Comment:
Note that, the previous wallet used to return this as a number of "confirmations" where confirmations referred to a number of _slot_. I recall that the wording was later change in our specification document by Ante for being more _correct_ at least conceptually. 

However, I'd call that an improvement over the existing behavior. For now, we can stick with computing the depth in number of `slots` and see later on whether we introduce or not, an extra metadata which is the number of blocks.

Needless metadata have more-or-less killed the previous wallet by introducing thousands of lines to simply compute a number we realized no one understood or cared about. Here is the same, it's unlikely client will understand without guidance the difference between slots and blocks. Plus, it's safer to stick with what they already know.

Reply:
Done: 37cf0b35

Reply:
I don't see any value in returning depth measured in slots. The client can calculate this themselves because they know the current slot, and the transaction's slot.

Transaction depth measured in blocks is a good measure of transaction confirmation. Depth measured in slots is possibly a bad measure of transaction confirmation.

It would be better to return no depth at all than a slot depth.

Reply:
At the time of designing the API, this was agreed as a "shortcut" / helper for the frontend though, I agree that I'd rather leave the calculation to the client and provide instead something that they can't actually compute.